src/stateMachines/featuresConfiguration.ts

```diff
@@ -0,0 +1,305 @@
+import { AMT, CIM, Common, IPS } from '@open-amt-cloud-toolkit/wsman-messages'
+import { HttpHandler } from '../HttpHandler'
+import Logger from '../Logger'
+import { assign, createMachine, interpret } from 'xstate'
+import { AMTConfiguration, AMTRedirectionServiceEnabledStates, mapAMTUserConsent } from '../models'
+import { devices } from '../WebSocketListener'
+import { ClientResponseMsg } from '../utils/ClientResponseMsg'
+import { parseBody } from '../utils/parseWSManResponseBody'
+import { AMT_REDIRECTION_SERVICE_ENABLE_STATE } from '@open-amt-cloud-toolkit/wsman-messages/models/common'
+import { RedirectionService } from '@open-amt-cloud-toolkit/wsman-messages/amt/models'
+import { IPS_OptInService } from '@open-amt-cloud-toolkit/wsman-messages/models/ips_models'
+
+export class FeaturesConfiguration {
+  amt: AMT.Messages
+  cim: CIM.Messages
+  ips: IPS.Messages
+  // the class name and method for this is silly
+  // should be new ClientReponseMsg(params...)
+  // but instead it has a method called .get?
+  clientMsgBuilder: ClientResponseMsg
+  httpHandler: HttpHandler
+  logger: Logger
+
+  constructor (
+    private readonly clientId: string,
+    private readonly amtCfg: AMTConfiguration
+  ) {
+    this.amt = new AMT.Messages()
+    this.cim = new CIM.Messages()
+    this.ips = new IPS.Messages()
+    this.clientMsgBuilder = new ClientResponseMsg()
+    this.httpHandler = new HttpHandler()
+    this.logger = new Logger('FeaturesConfiguration')
+  }
+
+  machine = createMachine({
+    id: 'features-configuration-fsm',
+    predictableActionArguments: true,
+    schema: {
+      context: {} as {
+        // 3 requests to the device
+        // to get the current configuration
+        // get saved here
+        AMT_RedirectionService?: any
+        IPS_OptInService?: any
+        CIM_KVMRedirectionSAP?: any
+        // transient values for computing
+        // what configuration changes should be made
+        // on the client device
+        isRedirectionChanged: boolean
+        isOptInServiceChanged: boolean
+        errorMessage: string
+      },
+      events: {} as {
+        type: ''
+        clientId: string
+        data?: any
+      }
+    },
+    initial: 'GET_AMT_REDIRECTION_SERVICE',
+    context: {
+      isRedirectionChanged: false,
+      isOptInServiceChanged: false,
+      errorMessage: ''
+    },
+    states: {
+      GET_AMT_REDIRECTION_SERVICE: {
+        invoke: {
+          src: async (context, _) => await this.getAmtRedirectionService(),
```

Comment:
Please update these `src:` properties to follow the same format we have in the other machines:
`src: this.getAmtRedirectionService.bind(this)`